modules/util/util.go

```diff
@@ -193,38 +192,6 @@ func ToTitleCase(s string) string {
 	return titleCaser.String(s)
 }
 
-var (
-	whitespaceOnly    = regexp.MustCompile("(?m)^[ \t]+$")
-	leadingWhitespace = regexp.MustCompile("(?m)(^[ \t]*)(?:[^ \t\n])")
-)
-
-// Dedent removes common indentation of a multi-line string along with whitespace around it
-// Based on https://github.com/lithammer/dedent
-func Dedent(s string) string {
-	var margin string
-
-	s = whitespaceOnly.ReplaceAllString(s, "")
-	indents := leadingWhitespace.FindAllStringSubmatch(s, -1)
-
-	for i, indent := range indents {
-		if i == 0 {
-			margin = indent[1]
-		} else if strings.HasPrefix(indent[1], margin) {
-			continue
-		} else if strings.HasPrefix(margin, indent[1]) {
-			margin = indent[1]
-		} else {
-			margin = ""
-			break
-		}
-	}
-
-	if margin != "" {
-		s = regexp.MustCompile("(?m)^"+margin).ReplaceAllString(s, "")
-	}
-	return strings.TrimSpace(s)
-}
-
```

Comment:
I'd like to keep it, it'll be useful for future tests.

Reply:
Also, I don't agree with having `[]string` instead of just a dedented multi-line string which is easier to edit.

Reply:
And one more benefit, when using `[]string`, we can use  `` ` `` string easliy

```
`<span class="n">b</span><span class="o">=</span><span class="sa"></span><span class="s1">&#39;</span><span class="s1">&#39;</span>`
```
VS
```
"
...
<span class=\"n\">b</span><span class=\"o\">=</span><span class=\"sa\"></span><span class=\"s1\">&#39;</span><span class=\"s1\">&#39;</span>
...
"
```

The new string code is more readable

Reply:
In commit https://github.com/silverwind/gitea/pull/2/commits/4e87838ae52db324913297ea871c568f606bee2b , I added a helper function `lines`, now the test string can be written in a single string, too.

Reply:
I'd still keep it. Test case `empty-line-with-space.py` would benefit by being more readable (no need to manually detent to 0 with it).

Reply:
I added the `Dedent` back. But I have no idea how to use it to improve the tests.

Feel free to edit/change it.

Reply:
Thanks, will give it a pass after merge.